src/backend/vulkan/src/device.rs

```diff
@@ -260,28 +287,34 @@ impl d::Device<B> for Device {
         n::PipelineLayout { raw }
     }
 
-    fn create_graphics_pipelines<'a>(
-        &self,
-        descs: &[pso::GraphicsPipelineDesc<'a, B>],
-    ) -> Vec<Result<n::GraphicsPipeline, pso::CreationError>> {
-        debug!("create_graphics_pipelines {:?}", descs);
+    fn create_graphics_pipelines<'a, T>(
+        &self, descs: T
+    ) -> Vec<Result<n::GraphicsPipeline, pso::CreationError>>
+    where
+        T: IntoIterator,
+        T::Item: Borrow<pso::GraphicsPipelineDesc<'a, B>>,
+    {
+        //debug!("create_graphics_pipelines {:?}", descs);
         const NUM_STAGES: usize = 5;
 
+        let descs = descs.into_iter();
+        let (lower_bound, upper_bound) = descs.size_hint();
+        let desc_len_guess = upper_bound.unwrap_or(lower_bound);
         // Store pipeline parameters to avoid stack usage
-        let mut info_stages                = Vec::with_capacity(descs.len());
-        let mut info_vertex_descs          = Vec::with_capacity(descs.len());
-        let mut info_vertex_input_states   = Vec::with_capacity(descs.len());
-        let mut info_input_assembly_states = Vec::with_capacity(descs.len());
-        let mut info_tessellation_states   = Vec::with_capacity(descs.len());
-        let mut info_viewport_states       = Vec::with_capacity(descs.len());
-        let mut info_rasterization_states  = Vec::with_capacity(descs.len());
-        let mut info_multisample_states    = Vec::with_capacity(descs.len());
-        let mut info_depth_stencil_states  = Vec::with_capacity(descs.len());
-        let mut info_color_blend_states    = Vec::with_capacity(descs.len());
-        let mut info_dynamic_states        = Vec::with_capacity(descs.len());
-        let mut color_attachments          = Vec::with_capacity(descs.len());
-        let mut info_specializations       = Vec::with_capacity(descs.len() * NUM_STAGES);
-        let mut specialization_data        = Vec::with_capacity(descs.len() * NUM_STAGES);
+        let mut info_stages                = Vec::with_capacity(desc_len_guess);
```

Comment:
Before I forget it over the holidays.. it crucial in the current implementation that these vectors **don't** reallocate memory because we have pointers into the vector elements floating around.